internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -95,27 +116,35 @@ func (r *BundleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ c
 	u.UpdateStatus(updater.EnsureObservedGeneration(bundle.Generation))
 
 	pod := &corev1.Pod{}
-	if op, err := r.ensureUnpackPod(ctx, bundle, pod); err != nil {
-		u.UpdateStatus(updater.SetBundleInfo(nil), updater.EnsureBundleDigest(""))
-		return ctrl.Result{}, updateStatusUnpackFailing(&u, fmt.Errorf("ensure unpack pod: %w", err))
-	} else if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated || pod.DeletionTimestamp != nil {
-		updateStatusUnpackPending(&u)
+	op, err := r.ensureUnpackPod(ctx, bundle, pod)
+	return r.transitionBundlePhase(BundlePhaseTransitioner{ctx, bundle, pod, op, &u, metricsRecorder, err})
+}
+
+func (r *BundleReconciler) transitionBundlePhase(pt BundlePhaseTransitioner) (_ ctrl.Result, reconcileErr error) {
+	// first record metric for phase bundle currently is in
+	pt.metricsRecorder.SetBundleMetric()
```

Comment:
Maybe I'm put off by the size of this PR at first glance and that's tainting my perception, but this seems like a somewhat heavyweight approach to simply incrementing/decrementing a single metric value.

There are two alternatives I have in mind, that I think would be much simpler
1. A totally separate controller whose only responsibility is to set metrics
2. A custom controller runtime handler that set metrics as events stream through the queue. We have something like that in `operator-lib`: https://github.com/operator-framework/operator-lib/blob/main/handler/instrumented_enqueue_object.go

Reply:
I have to look more closely at `operator-lib` and that particular handler, but before I spend time doing that, I'm curious to read the responses in the #266